internal/controller/operatorconfigmap_controller.go

```diff
@@ -469,21 +472,42 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI(storageClients *v1al
 				DomainLabels: slices.Collect(maps.Keys(topologyDomainLablesSet)),
 			}
 		}
+		driverSpecDefaults.GenerateOMapInfo = ptr.To(c.shouldGenerateRBDOmapInfo())
 		return nil
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile csi operator config: %v", err)
 	}
 
+	// get the custom csi Tolerations
+	// TODO: Remove it in 4.20 release
+	customNodePluginTolerations, customControllerPluginTolerations, err := c.getCsiCustomTolerations()
+	if err != nil {
+		return fmt.Errorf("failed to get custom tolerations for csi: %v", err)
+	}
+
 	// ceph rbd driver config
 	rbdDriver := &csiopv1a1.Driver{}
 	rbdDriver.Name = templates.RBDDriverName
 	rbdDriver.Namespace = c.OperatorNamespace
-
 	if err := c.createOrUpdate(rbdDriver, func() error {
 		if err := c.own(rbdDriver); err != nil {
 			return fmt.Errorf("failed to own csi rbd driver: %v", err)
 		}
-		rbdDriver.Spec.GenerateOMapInfo = ptr.To(c.shouldGenerateRBDOmapInfo())
+		// only update during initial creation
+		if rbdDriver.UID == "" {
```

Comment:
Your condition should be based on the UID (as it is here) and the existence of custom tolerations (missing check)